### PR TITLE
AWS test prep for WIF work

### PR DESCRIPTION
**Pull request by Monkeychip**
### Description
This pr adds and cleans up test coverage for the AWS secret engine configuration flow. This work will clarify what is going to change in the AWS WIF work, while also trying to prevent unknown regressions.

No user facing changes.

- [ ] enterprise test pass.

### CODE REVIEW

File: ui/tests/acceptance/secrets/backend/aws/aws-configuration-test.js [hunk 1]
```diff
@@ -77,15 +78,12 @@ module('Acceptance | aws | configuration', function (hooks) {
   });
 
   test('it should save root AWS configuration', async function (assert) {
-    assert.expect(3);
```
Comment (Monkeychip) [inline ui/tests/acceptance/secrets/backend/aws/aws-configuration-test.js:80]: removing assert.expect unless using mirage. To my knowledge these are only helpful if an assert could go unchecked. this use case only seems to happen when using something like `this.server.get` and the get is never hit.

File: ui/app/components/secret-engine/configure-aws.hbs [hunk 2]
```diff
@@ -16,7 +16,7 @@
         "saveRootCreds"
         (hash access_key=@accessKey iam_endpoint=@iamEndpoint sts_endpoint=@stsEndpoint secret_key=@secretKey region=@region)
       }}
-      data-test-aws-root-creds-form="true"
+      data-test-root-form
```
Comment (Monkeychip) [inline ui/app/components/secret-engine/configure-aws.hbs:19]: making selectors consistent across ssh and aws configuration components.

File: ui/tests/acceptance/secrets/backend/aws/aws-test.js [hunk 3]
```diff
@@ -97,8 +97,8 @@ module('Acceptance | aws secret backend', function (hooks) {
 
     // delete role
     await click(`${SES.secretLink(roleName)} [data-test-popup-menu-trigger]`);
-    await waitUntil(() => find(SES.aws.delete(roleName))); // flaky without
```
Comment (Monkeychip) [inline ui/tests/acceptance/secrets/backend/aws/aws-test.js:100]: name change for clarity on SES selectors.

**Review by hashishaw — APPROVED** [review]
Nice, thanks for the extra test coverage!

**Pull request merged**